docs/arch/controller.md

```diff
@@ -0,0 +1,40 @@
+# Architecture Document for Controller Component
+
+```mermaid
+sequenceDiagram
+    actor User
+    User->>Controller: sends pipeline jobs as YAML and repository link
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+    Controller->>Database: saves pipeline in database
+    loop over steps
+        Controller-->>Scheduler: sends step (over GRPC)
+        Scheduler-->>Controller: job succeeded or not
+    end
+    User->>Controller: get pipeline state
+```
+
+## Glossary
+- action : list of steps, a declarative file can only contain one action.
```

Comment:
What is a "declarative file" ? one "action" is contained in each one of these files ?